mlflow/gateway/providers/bedrock.py

```diff
@@ -0,0 +1,279 @@
+from mlflow.gateway.providers.base import BaseProvider
+from mlflow.gateway.providers.utils import rename_payload_keys, send_request
+from mlflow.gateway.schemas import chat, completions, embeddings
+
+
+class AWSBedrockModelProvider(Enum):
+    AMAZON = "amazon"
+    COHERE = "cohere"
+    AI21 = "ai21"
+    ANTHROPIC = "anthropic"
+
+    @classmethod
+    def of_str(cls, name: str):
+        name = name.lower()
+
+        for opt in cls:
+            if opt.name.lower() == name or opt.value.lower() == name:
+                return opt
+
+
+class AWSBedrockProvider(BaseProvider):
+    def __init__(self, config: RouteConfig):
+        super().__init__(config)
+
+        if config.model.config is None or not isinstance(config.model.config, AWSBedrockConfig):
+            raise TypeError(f"Invalid config type {config.model.config}")
+        self.bedrock_config: AWSBedrockConfig = config.model.config
+        self._client = None
+        self._client_created = 0
+
+    def _client_expired(self):
+        if not isinstance(self.bedrock_config.aws_config, AWSRole):
+            return False
+
+        return (
+            (time.monotonic_ns() - self._client_created)
+            >= (self.bedrock_config.aws_config.session_length_seconds) * 1_000_000_000,
+        )
+
+    def get_bedrock_client(self):
+        if self._client is not None and not self._client_expired():
+            return self._client
+
+        session_args, config_args, client_args = {}, {}, {}
+        aws_config = self.bedrock_config.aws_config
+
+        region_name = aws_config.aws_region or os.environ.get(
+            "AWS_REGION", os.environ.get("AWS_DEFAULT_REGION")
+        )
+
+        if region_name:
+            config_args["region_name"] = session_args["region_name"] = region_name
+
+        # check profile, mostly for dev
+        profile_name = os.environ.get("AWS_PROFILE")
+        if profile_name:
+            session_args["profile_name"] = profile_name
+
+        session = boto3.Session(**{k: v for k, v in session_args.items() if v is not None})
+
+        if isinstance(aws_config, AWSRole):
+            role = session.client(service_name="sts").assume_role(
+                RoleArn=aws_config.aws_role_arn,
+                RoleSessionName="ai-gateway-bedrock",
+                DurationSeconds=aws_config.session_length_seconds,
+            )
+            client_args["aws_access_key_id"] = role["Credentials"]["AccessKeyId"]
+            client_args["aws_secret_access_key"] = role["Credentials"]["SecretAccessKey"]
+            client_args["aws_session_token"] = role["Credentials"]["SessionToken"]
+        elif isinstance(aws_config, AWSIdAndKey):
+            client_args["aws_access_key_id"] = aws_config.aws_access_key_id
+            client_args["aws_secret_access_key"] = aws_config.aws_secret_access_key
+            client_args["aws_session_token"] = aws_config.aws_session_token
+        else:
+            # defer to standard CredentialProviderChain
```

Comment:
If there are issues with the CredentialProviderChain, what sort of Exception is raised? Will the server pass this issue correctly back to the client with an informative message about why a 5xx error occurred within the AI Gateway?